graylog2-server/src/main/java/org/graylog2/migrations/V20250917184400_AddSystemEntitySource.java

```diff
@@ -0,0 +1,125 @@
+ */
+package org.graylog2.migrations;
+
+import com.mongodb.client.model.Filters;
+import com.mongodb.client.model.InsertOneModel;
+import jakarta.inject.Inject;
+import org.bson.conversions.Bson;
+import org.graylog.events.processor.DBEventDefinitionService;
+import org.graylog.events.processor.EventDefinitionDto;
+import org.graylog2.contentpacks.ContentPackInstallationPersistenceService;
+import org.graylog2.contentpacks.model.ContentPackInstallation;
+import org.graylog2.database.MongoCollection;
+import org.graylog2.database.MongoCollections;
+import org.graylog2.database.entities.NonDeletableSystemScope;
+import org.graylog2.database.entities.source.DBEntitySourceService;
+import org.graylog2.database.entities.source.EntitySource;
+import org.graylog2.plugin.cluster.ClusterConfigService;
+
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static org.graylog2.contentpacks.model.ModelTypes.DASHBOARD_V1;
+import static org.graylog2.contentpacks.model.ModelTypes.DASHBOARD_V2;
+import static org.graylog2.contentpacks.model.ModelTypes.SEARCH_V1;
+import static org.graylog2.database.utils.MongoUtils.stream;
+
+public class V20250917184400_AddSystemEntitySource extends Migration {
+
+
+    private final ClusterConfigService configService;
+    private final MongoCollection<EventDefinitionDto> eventDefinitionCollection;
+    private final MongoCollection<EntitySource> entitySourceCollection;
+    private final com.mongodb.client.MongoCollection<ContentPackInstallation> contentPackCollection;
+
+    @Inject
+    public V20250917184400_AddSystemEntitySource(ClusterConfigService configService,
+                                                 MongoCollections mongoCollections) {
+        this.configService = configService;
+        this.eventDefinitionCollection = mongoCollections.collection(DBEventDefinitionService.COLLECTION_NAME, EventDefinitionDto.class);
+        this.entitySourceCollection = mongoCollections.collection(DBEntitySourceService.COLLECTION_NAME, EntitySource.class);
+        this.contentPackCollection = mongoCollections.nonEntityCollection(ContentPackInstallationPersistenceService.COLLECTION_NAME, ContentPackInstallation.class);
+    }
+
+    @Override
+    public ZonedDateTime createdAt() {
+        return ZonedDateTime.parse("2025-09-17T18:44:00Z");
+    }
+
+    @Override
+    public void upgrade() {
+        if (migrationAlreadyApplied()) {
+            return;
+        }
+
+        insertSystemEntitySourceDocuments();
+
+        markMigrationApplied();
+    }
+
+    private void insertSystemEntitySourceDocuments() {
+        final List<InsertOneModel<EntitySource>> entitySourceRecords = new ArrayList<>();
+
+        final V20230601104500_AddSourcesPageV2.MigrationCompleted sourcesMigration = configService.get(V20230601104500_AddSourcesPageV2.MigrationCompleted.class);
```

Comment:
I think it would be worth adding a comment in the Sources dash create migration (and the [System events def](https://github.com/Graylog2/graylog2-server/blob/4309a7dc727f351fe83f83789340620514e0ec56/graylog2-server/src/main/java/org/graylog2/migrations/V20190705071400_AddEventIndexSetsMigration.java#L210) migration) that the source is added separately in this migration just as an FYI.

Reply:
Probably a good idea in case anyone uses those migrations as a basis for installing system-provided content in the future. I have left the comments in the most recent commit.